modules/core/nuxeo-core-storage-mem/src/test/java/org/nuxeo/ecm/core/storage/mem/TestMemRepositoryService.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * (C) Copyright 2021 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Anahide Tchertchian
+ */
+package org.nuxeo.ecm.core.storage.mem;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.core.api.repository.Repository;
+import org.nuxeo.ecm.core.api.repository.RepositoryManager;
+import org.nuxeo.ecm.core.storage.dbs.DBSRepositoryDescriptor;
+import org.nuxeo.ecm.core.storage.dbs.DBSRepositoryService;
+import org.nuxeo.runtime.cluster.ClusterFeature;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.RuntimeFeature;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+
+/**
+ * @since 11.5
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ RuntimeFeature.class, ClusterFeature.class, TransactionalFeature.class })
+@Deploy("org.nuxeo.ecm.core:OSGI-INF/RepositoryService.xml")
+@Deploy("org.nuxeo.ecm.core.schema")
+@Deploy("org.nuxeo.ecm.core.storage")
+@Deploy("org.nuxeo.ecm.core.storage.dbs")
+@Deploy("org.nuxeo.ecm.core.storage.mem")
+@Deploy("org.nuxeo.ecm.core.storage.mem:mock-repo-manager-contrib.xml")
+public class TestMemRepositoryService {
+
+    @Inject
+    protected MemRepositoryService memRepositoryService;
+
+    @Inject
+    protected DBSRepositoryService dbsService;
+
+    @Inject
+    protected RepositoryManager repositoryManager;
+
+    /**
+     * Checks correct injection of the service.
+     */
+    @Test
+    public void testService() {
+        assertNotNull(memRepositoryService);
+    }
```

Comment:
Indeed, this one actually comes from a copy of the `TestMongoDBRepositoryService#testUP` that I did not clean up, I will do so.

Reply:
And, related: when hot-reloading during tests, the injected components are sometimes kept but with empty fields, this is sometimes confusing... (this would need to be fixed at the injection system level I guess)